File: Project-Frontend - Copy/digital_library_app/lib/models/book.dart
class Book {
  final int id;
  final String title;
  final String? author;
  final String? category;
  final String? coverImageUrl;
  final String? pdfUrl;

  Book({
    required this.id,
    required this.title,
    this.author,
    this.category,
    this.coverImageUrl,
    this.pdfUrl,
  });

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: (json['id'] as num).toInt(),
      title: (json['title'] ?? '').toString(),
      author: json['author']?.toString(),
      category: json['category']?.toString(),
      coverImageUrl: json['coverImageUrl']?.toString(),
      pdfUrl: json['pdfUrl']?.toString(),
    );
  }
}

File: digital-library/src/main/java/com/library/digitallibrary/book/Book.java
package com.library.digitallibrary.book;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.library.digitallibrary.review.Review;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "books")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    private String author;

    @Column(nullable = false)
    private String category;

    // cover image URL/path
    private String coverImageUrl;

    // PDF URL/path
    private String pdfUrl;

    @OneToMany(mappedBy = "book", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<Review> reviews = new ArrayList<>();

    @OneToMany(mappedBy = "book", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<Favorite> favorites = new ArrayList<>();

    @OneToMany(mappedBy = "book", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<ReadLater> readLaters = new ArrayList<>();
}
